verilog/analysis/verilog_linter.h

```diff
@@ -34,19 +35,44 @@
 
 namespace verilog {
 
+// Base class of LintRuleStatus handlers
+class LintStatusHandler {
+ public:
+  LintStatusHandler() {}
+
+  // Handles LintRuleStatuses. By default sorts all violations by position of
+  // their occurence in source file and passes them to HandleViolation() method
+  // in a loop.
+  // This method can be called multiple times with statuses generated from
+  // different files.
+  virtual void HandleLintRuleStatuses(
+      const std::vector<verible::LintRuleStatus>& statuses,
+      absl::string_view base, absl::string_view path);
+
+ protected:
+  // Handles single violation.
+  virtual void HandleViolation(const verible::LintViolation& violation,
+                               absl::string_view base, absl::string_view path,
+                               absl::string_view url,
+                               absl::string_view rule_name) {}
+};
+
 // Checks a single file for Verilog style lint violations.
 // This is suitable for calling from main().
-// Diagnostics are printed to 'stream'.
+// 'stream' is used for printing potential syntax errors (if 'check_syntax' is
+// true).
 // 'filename' is the path to the file to analyze.
 // 'config' controls lint rules for analysis.
+// 'lint_status_handler' controls what to do with violations.
 // If 'check_syntax' is true, report lexical and syntax errors.
 // If 'parse_fatal' is true, abort after encountering syntax errors, else
 // continue to analyze the salvaged code structure.
 // If 'lint_fatal' is true, exit nonzero on finding lint violations.
 // Returns an exit_code like status where 0 means success, 1 means some
 // errors were found (syntax, lint), and anything else is a fatal error.
 int LintOneFile(std::ostream* stream, absl::string_view filename,
-                const LinterConfiguration& config, bool check_syntax,
+                const LinterConfiguration& config,
+                LintStatusHandler& lint_status_handler, bool check_syntax,
```

Comment:
Always pass something we modify as a pointer, never as non-const reference, so this should be `LintStatusHandler*`.

Reply:
Done